src/indexer/segment_updater.rs

```diff
@@ -199,13 +189,61 @@ pub fn merge_segments<Dir: Directory>(
         segments.extend(index.searchable_segments()?);
     }
 
-    let mut merged_index = Index::create(output_directory, target_schema.clone(), target_settings)?;
+    let non_filter = segments.iter().map(|_| None).collect::<Vec<_>>();
+    merge_filtered_segments(&segments, target_settings, non_filter, output_directory)
+}
+
+/// Advanced: Merges a list of segments from different indices in a new index.
+/// Additional you can provide a delete bitset for each segment to ignore docids.
+///
+/// Returns `TantivyError` if the the indices list is empty or their
+/// schemas don't match.
+///
+/// `output_directory`: is assumed to be empty.
+///
+/// # Warning
+/// This function does NOT check or take the `IndexWriter` is running. It is not
+/// meant to work if you have an IndexWriter running for the origin indices, or
+/// the destination Index.
+#[doc(hidden)]
+pub fn merge_filtered_segments<Dir: Directory>(
```

Comment:
The generics is useless here.

Reply:
I added it since merge_indices also uses it, and they are basically the same code. Is it also useless there?

`pub fn merge_indices<Dir: Directory>(`